script/fetch-issues.js

```diff
@@ -141,23 +192,132 @@ async function fetchNextPage(owner, name, isIssue, data, cursor, pageCount) {
         }
       };
       if (!isIssue) {
-        info.pullRequest = {
-          htmlUrl: node.url
+        let rawComments, comments;
+        // collect review comments
+        rawComments = node.reviews.nodes.map(node => node.comments.nodes);
+        rawComments = [].concat.apply([], rawComments);
+        // collect issue comments
+        rawComments = rawComments.concat(node.comments.nodes);
+
+        // fetch reactions only if there are reactions within that PR
+        // to reduce API hits.
+        // trick: a comment has received reaction(s) iff
+        // reactionGroup.createdAt attribute is not null
+        let commentsWithReactions = [];
+        const hasReactions = rawComments.reduce((has, rawComment) => {
+          return has || rawComment.reactionGroups.reduce((flag, node) => {
+            return (flag || node.createdAt) ? true : false;
+          }, false);
+        }, false);
+
+        if (hasReactions) {
+          console.log('pull request has reactions. repoOwner:', owner,
+            'repoName:', name, 'pullRequest number:', node.number);
+          const number = node.number;
+          const reviewCnt = Math.min(node.reviews.totalCount, 20);
+          const maxCommentsPerReview = Math.min(
+            node.reviews.nodes.reduce(
+              (max, review) => Math.max(max, review.comments.totalCount), 0),
+            100);
+          const commentCnt = Math.min(node.comments.totalCount, 100);
+          commentsWithReactions = await fetchReactionsOfPR(owner, name, number,
+            reviewCnt, maxCommentsPerReview, commentCnt);
+
+          // merge rawComments with commentsWithReactions
+          if (commentsWithReactions) {
+            rawComments.map((comment, index) => {
+              if (comment.id !== commentsWithReactions[index].id) {
+                console.log('warning: comments and reactions do not fit!',
+                  'pr number:', number, 'comment.id:', comment.id,
+                  'comment with reactions id:', commentsWithReactions[index].id);
+              } else {
+                comment.reactions = commentsWithReactions[index].reactions;
+              }
+              return comment;
+            });
+          }
+        } else {
+          // for debug
+          // console.log('pull request does not have reactions. repoOwner:', owner,
```

Comment:
If this is really necessary, it's better to have a `DEBUG` env var that enable/disable verbose log, otherwise just omit.